fuzzers/qemu_launcher/src/fuzzer.rs

```diff
@@ -64,53 +131,151 @@ pub fn fuzz() {
         .expect("Symbol LLVMFuzzerTestOneInput not found");
     println!("LLVMFuzzerTestOneInput @ {test_one_input_ptr:#x}");
 
-    emu.set_breakpoint(test_one_input_ptr); // LLVMFuzzerTestOneInput
+    emu.set_breakpoint(test_one_input_ptr);
     unsafe { emu.run() };
 
-    println!("Break at {:#x}", emu.read_reg::<_, u64>(Regs::Rip).unwrap());
+    for m in emu.mappings() {
+        println!(
+            "Mapping: 0x{:016x}-0x{:016x}, {}",
+            m.start(),
+            m.end(),
+            m.path().unwrap_or("<EMPTY>")
+        );
+    }
+
+    let read_reg = |emu: &Emulator, reg: Regs| -> GuestReg {
+        let val: GuestReg = emu.read_reg(reg).unwrap();
+
+        #[cfg(feature = "be")]
+        return GuestReg::from_be(val);
 
-    // Get the return address
-    let stack_ptr: u64 = emu.read_reg(Regs::Rsp).unwrap();
-    let mut ret_addr = [0; 8];
-    unsafe { emu.read_mem(stack_ptr, &mut ret_addr) };
-    let ret_addr = u64::from_le_bytes(ret_addr);
+        #[cfg(not(feature = "be"))]
+        return GuestReg::from_le(val);
+    };
+
+    let write_reg = |emu: &Emulator, reg: Regs, val: GuestReg| {
+        #[cfg(feature = "be")]
+        let val = GuestReg::to_be(val);
+
+        #[cfg(not(feature = "be"))]
+        let val = GuestReg::to_le(val);
+
+        emu.write_reg(reg, val).unwrap();
+    };
+
+    println!("Break at {:#x}", read_reg(&emu, Regs::Pc));
+
+    #[cfg(feature = "arm")]
```

Comment:
We could move this to libafl_qemu as `read_ret_addr` or similar

Reply:
Could do the same with register reading and endian conversion? Is it worthwhile trying to push the architecture dependent stuff down into libafl_qemu as much as possible?

Reply:
It might be cool to have these common features in the library, I think, right? But also this one is more a general idea and not direclty related to this PR :)

CC @andreafioraldi